misc-scripts/xref_mapping/XrefParser/UniProtParser.pm

```diff
@@ -459,6 +464,21 @@ sub create_xrefs {
             $num_direct_sptr++;
           }
           push @{$xref->{DIRECT_XREFS}}, \%direct;
+
+          my $uniprot_acc = $accessions[0];
+          if ($extra[1] =~ /($accessions[0]-[0-9]+)/) {
+            $isoform = $1;
```

Comment:
It's quite likely I'm missing something here, but I'm wondering if `$isoform` will be `[P31946-1]` with the literal curly braces included at this point?

Reply:
$extra[1] looks like `ENSG00000159840. [Q15942-1]`
we are extracting the part that starts with `$accessions[0]` and ends with some digits, curly braces are not included in the match
once parsed, the xrefs look like this in the database:
```
115171	P31946-1	0	P31946-1	NULL	141	9606	DIRECT		NULL
115189	P62258-1	0	P62258-1	NULL	141	9606	DIRECT		NULL
115193	P62258-2	0	P62258-2	NULL	141	9606	DIRECT		NULL
```

Reply:
Of course, it's just matching the accession. Brain stall on my part.